test/built-ins/Array/prototype/fill/return-abrupt-from-setting-property-value.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.1.3.6
+description: >
+  Return abrupt from setting a property value.
+info: >
+  22.1.3.6 Array.prototype.fill (value [ , start [ , end ] ] )
+
+  ...
+  11. Repeat, while k < final
+    a. Let Pk be ToString(k).
+    b. Let setStatus be Set(O, Pk, value, true).
+    c. ReturnIfAbrupt(setStatus).
+  ...
+---*/
+
+var a1 = [];
+Object.freeze(a1);
+
+// won't break on an empty array.
+a1.fill(1);
+
+var a2 = [0];
+Object.freeze(a2);
```

Comment:
`Object.freeze` makes sense for the first case because it ensures that there is no property setting at all. But for the negative case, we should be more specific about exactly *which* operation we expect to take place. You could do this by defining `0` as an accessor property that throws a `Test262Error` when set.